src/sentry/api/endpoints/project_codeowners.py

```diff
@@ -61,6 +64,16 @@ def validate(self, attrs: Mapping[str, Any]) -> Mapping[str, Any]:
         if not attrs.get("raw", "").strip():
             return attrs
 
+        # We want to limit `raw` to a reasonable length, so that people don't end up with values
+        # that are several megabytes large. To not break this functionality for existing customers
+        # we temporarily allow rows that already exceed this limit to still be updated.
+        # We do something similar with ProjectOwnership at the API level.
+        existing_raw = self.instance.raw or ""
+        if len(attrs["raw"]) > MAX_RAW_LENGTH and len(existing_raw) <= MAX_RAW_LENGTH:
```

Comment:
Just a thought,
In order to get have the raw codeowners file, it would have to already be loaded into memory before we do the length check. If we're trying to protect ourselves with this check, we should be preventing the upload to begin with from the frontend.

Reply:
@AniketDas-Tekky yea, but I think the issue is not memory but the CPU overhead from calculating the assignments on each event.